### PR TITLE
Accept POST requests with JSON body

### PR DESCRIPTION
**Pull request by sveinbjornt**
* The correct.api endpoint now supports JSON body in POST request (#6):

The JSON should have the format:
````
{
    "text": "Textinn til að lesa yfir."
}
````
````
curl -d @example.json https://yfirlestur.is/correct.api --header "Content-Type: application/json"
````

* The correct.api endpoint now supports several configurable options: `annotate_unparsed_sentences` (bool), `suppress_suggestions` (bool), `ignore_wordlist` (list), `ignore_rules` (list). These options can be provided either via a www form encoded POST request or a POSTed JSON dict :
````
{
    "text": "...",
    "annotate_unparsed_sentences": false,
    "suppress_suggestions": true,
    "ignore_wordlist": ["florg", "smorg", "glorg"],
    "ignore_rules": ["P_WRONG_CASE"]
}
````
````
curl -d @example.json https://yfirlestur.is/correct.api --header "Content-Type: application/json"
````
* Many tests added for the abovementioned functionality
* Various other Flask test client tests to verify that all web routes are functioning correctly
* About page now shows Tokenizer, Greynir version + platform name, like https://greynir.is/about
* Updated documentation to reflect these changes
* Various linter fixes and formatting (black + others)

### CODE REVIEW

**Comment by sveinbjornt** [comment]
@Holado @vthorsteinsson This is ready for review. 

By the way, are the names of rules (for ignore_rules param) documented anywhere except in the GreynirCorrect codebase?

File: routes/api.py [hunk 1]
```diff
@@ -235,44 +239,71 @@ def feedback(version: int = 1) -> Any:
     return better_jsonify(ok=True)
 
 
+OPTIONS_FLAGS = {
```
Comment (vthorsteinsson) [inline routes/api.py:242]: Annotate: Mapping[str, str]

File: routes/api.py [hunk 2]
```diff
@@ -235,44 +239,71 @@ def feedback(version: int = 1) -> Any:
...
+    "annotate_unparsed_sentences": "bool",
+    "suppress_suggestions": "bool",
+    "ignore_wordlist": "list",
+    "ignore_rules": "list",
+}
+
+
+def opts_from_request(rq: Request) -> Dict[str, Any]:
+    """Extract valid options from a request into dict."""
+    d = dict()
```
Comment (vthorsteinsson) [inline routes/api.py:252]: Annotate: d: Dict[str, Any] = dict()

...and strictly speaking, the annotation could (and should?) be made accurate:

`OptionsDict = Dict[str, Union[bool, List[str]]]`

File: templates/about.html [hunk 3]
```diff
@@ -157,5 +167,4 @@
 
 </script>
 
-{% endblock %}
-
+{% endblock %}
```
Comment (vthorsteinsson) [inline templates/about.html:170]: Fix line ending?

**Comment by Holado** [comment]
> By the way, are the names of rules (for ignore_rules param) documented anywhere except in the GreynirCorrect codebase?
@sveinbjornt as mentioned, they are added in PR https://github.com/mideind/GreynirCorrect/pull/48 and will be available shortly. Documentation within Yfirlestur pointing to them is ready on a separate branch.

**Pull request merged**